did_core/did_parser_nom/src/did/parsing/did_web.rs

```diff
@@ -0,0 +1,22 @@
+use nom::{
+    bytes::complete::{tag, take_till},
+    character::complete::char,
+    combinator::{fail, opt},
+    sequence::{delimited, tuple},
+    IResult,
+};
+
+use super::DidPart;
+
+pub(super) fn parse_did_web(input: &str) -> IResult<&str, DidPart> {
+    fn did_web_method(input: &str) -> IResult<&str, &str> {
+        delimited(char(':'), tag("web"), char(':'))(input)
+    }
+
+    tuple((
+        tag("did"),
+        did_web_method,
+        opt(fail::<_, &str, _>),
+        take_till(|c| "?/#".contains(c)),
```

Comment:
A comment could be nice, this is non-obvious for reader. You mentioned there's some trickiness with `did:web`, but still I am not sure what this is/why

Reply:
This stops parsing the id portion when either DID URL query, path or fragment are encountered.

Reply:
I understand that `did:web` can have variable number of colons, eg `did:web:w3c-ccg.github.io:user:foo:bar:alice`, but I am still missing the `?/#` handling here. If I understand the current implementation right, given input 
```
did:web:w3c-ccg.github.io:user:alice#abcd
```
which is in fact DID URL, will in fact succesfully parse DID `did:web:w3c-ccg.github.io:user:alice`. Is that right?

But if `did:web` parser implementation is lenient enough to accept DID URL as DID, why wouldn't other did methods do the same? Eg. shouldn't `did:sov:abcd123#abcd` be also parsed as DID `did:sov:abcd123`.

Reply:
All the parsers parse DID URLs (to be reusable when parsing a DID URL), but for DID parsing the `all_consuming` combinator is used.

Reply:
I think it would be worthy a brief comment in the code, but no biggie